### PR TITLE
ref(notification platform): Add team scope type

### PR DESCRIPTION
**Pull request by ceorourke**
**Context**
As we're building out linking a team to a Slack channel to receive notifications in we realized we overlooked having a project agnostic team scope type which would allow us to turn on / off notifications for all of a team's projects. 

Also, we don't currently have designs for team settings that would allow a team to opt out of notifications for individual projects, but it's fairly likely that we will want that in the future. This design still allows for that given we can pass a project into the notification settings manager with the actor as a team. 

Add `NotificationSettingScopeType.TEAM` to the `NotificationSetting` `scope_type` field.

### CODE REVIEW

File: src/sentry/migrations/0208_add_team_scope.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        ("sentry", "0207_release_adoption"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="notificationsetting",
+            name="scope_type",
+            field=sentry.db.models.fields.bounded.BoundedPositiveIntegerField(
+                choices=[
+                    (sentry.notifications.types.NotificationScopeType(30), "team"),
```
Comment (markstory) [inline src/sentry/migrations/0208_add_team_scope.py:39]: Does this no longer need to list all the choices?

Comment (ceorourke) [reply src/sentry/migrations/0208_add_team_scope.py:39]: oh oops! good catch I will list all of them

**Review by mgaeta — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Review by wedamija — COMMENTED** [review]
Hmm, I thought I modified migrations so they wouldn't care about changes to choices...

**Comment by wedamija** [comment]
> Hmm, I thought I modified migrations so they wouldn't care about changes to choices...

@ceorourke I tested this out locally, added a new type to scope and ran `makemigrations` and it didn't generate anything, wondering if I did anything different to you?

**Comment by ceorourke** [comment]
@wedamija I experienced the same but thought that it meant I had to write a custom migration 😬 is having this bad?

**Comment by wedamija** [comment]
> @wedamija I experienced the same but thought that it meant I had to write a custom migration 😬 is having this bad?

Nope, won't cause any issues, I just wanted to make sure my changes hadn't broken. But you don't need to worry about writing migrations for choice fields, I made Django not care about them here: https://github.com/getsentry/sentry/pull/26346

**Pull request merged**